src/hu_stuff.c

```diff
@@ -1010,15 +1011,16 @@ static void HU_InitCrosshair(void)
 
     crosshair.w = SHORT(crosshair.patch->width)/2;
     crosshair.h = SHORT(crosshair.patch->height)/2;
-    crosshair.x = ORIGWIDTH/2;
   }
   else
     crosshair.patch = NULL;
 }
 
 static void HU_UpdateCrosshair(void)
 {
+  crosshair.x = ORIGWIDTH/2;
   crosshair.y = (screenblocks <= 10) ? (ORIGHEIGHT-ST_HEIGHT)/2 : ORIGHEIGHT/2;
+  crosshair.lockx = crosshair.locky = 0;
```

Comment:
If we are setting the `x` / `y` here in the update code, do we really need the `lockx` / `locky` variables? As in, can we just do `crosshair.x += x` in `HU_UpdateCrosshairLock`?

Reply:
I tried what you're suggesting before, but doing that causes the crosshair to jitter. It would appear that `R_ProjectSprite()` for `linetarget` is called more than once before `HU_UpdateCrosshair()` is called again.

Reply:
> It would appear that `R_ProjectSprite()` for [`crosshair_target`] is called more than once before `HU_UpdateCrosshair()` is called again.

Did a bit of testing, and this does seem to be the cause of the issue. As I expected, it only happens when Uncapped Framerate is enabled.
Knowing this, I tried something: by setting `crosshair_target = NULL` right after calling `HU_UpdateCrosshairLock()` (within the same code block), the crosshair doesn't jitter when directly adding to `x` and `y`. This effectively makes the `lock#` variables redundant, so I could remove them. @fabiangreffrath, do you want me to do this?

Reply:
Sure, sounds reasonable, thanks for the idea!